oasis/models/rules.py

```diff
@@ -2,15 +2,167 @@
+                                              rule=rule), self.job.slack_channel)
+                        not_match_count = not_match_count + 1
+
+                if not_match_count == 0:
+                    logger.info("[job-id:{id}][metric:{metric}] predict OK"
+                                .format(id=sub_id(self.job.id), metric=metric))
+
+            self.event.wait(timeparse(self.cfg.model["predict_interval"]))
+
+        logger.info("[job-id:{id}][metric:{metric}] stop"
+                    .format(id=sub_id(self.job.id), metric=metric))
+
+    def close(self):
+        logger.info("[job-id:{id}] closing the job"
+                    .format(id=sub_id(self.job.id)))
+        super(Rules, self).close()
+        self.timer.cancel()
+
+    def timeout_action(self):
+        logger.info("[job-id:{id}] finish the job"
+                    .format(id=sub_id(self.job.id)))
+        super(Rules, self).close()
+
+
+class RulesConfig(Config):
+    def __init__(self, config_file, config=None):
+        super(RulesConfig, self).__init__()
+
+        self.rules = dict()
+        self._init(config_file, config)
+        self._parse_rules()
+
+    def _set_default_config(self):
+        self.model.setdefault("data_range", "10m")
+        self.model.setdefault("predict_interval", "5m")
+
+        for metric in self.metrics:
+            self.metrics[metric].setdefault("features", ["mean", "std"])
+            self.metrics[metric].setdefault("rules",
+                                            ['features["std"] > 1000'])
+
+    def _parse_rules(self):
+        for metric in self.metrics:
+            rules = self.metrics[metric].get("rules")
+            rules_list = []
+            for rule in rules:
+                index_left = rule.find("[")
+                if index_left == -1:
+                    logger.info("rule:{rule} is invalid, skip it"
+                                .format(rule=rule))
+
+                index_right = rule.find("]")
+                if index_right == -1:
+                    logger.info("rule:{rule} is invalid, skip it"
+                                .format(rule=rule))
+
+                feature = rule[index_left+1:index_right]
+
+                operator = ""
+                operator_index = -1
+                for op in OPERATORS:
+                    operator_index = rule.find(op)
+                    if operator_index != -1:
+                        operator = op
+                        break
+
+                if operator_index == -1:
+                    continue
+
+                value_str = rule[operator_index+len(operator):]
+                value = float(value_str)
+                rules_list.append({"expr": rule, "feature": feature,
+                                   "operator": operator, "value": value})
 
+            self.rules[metric] = rules_list
 
 
+def check_is_triggered(left_value, operator, right_value):
+    return {
```

Comment:
suggestion: all keys are evaluated eagerly when the dict is initialized. This is costy if evaluation is expensive.

You can try something like `"==": lambda: abs(left_value-right_value) < 0.0001`, and `.get(operator, lambda: False)()`.